krod/core/llm_manager.py

```
@@ -627,26 +633,22 @@ def _generate_openai(self, prompt: str, model: str, temperature: float, max_toke
-            error_msg = "Request timed out while waiting for response"
-            self.logger.error(error_msg)
+            async with aiohttp.ClientSession() as session:
+                async with session.post(url, headers=headers, json=data, timeout=aiohttp.ClientTimeout(total=30)) as resp:
+                    if resp.status == 200:
+                        result = await resp.json()
+                        return result["choices"][0]["message"]["content"]
+                    else:
+                        text = await resp.text()
+                        self.logger.error(f"OpenAI API error: {resp.status} - {text}")
+                        return "I apologize, but I encountered an error processing your request."
+        except asyncio.TimeoutError:
```

Flake8 (7.2.0)

[error] undefined name 'asyncio'

(F821)